### PR TITLE
SNOW-1662275: Add support for parameters `left_index` and `right_index` for `pd.merge_asof`

### PR DESCRIPTION
**Pull request by sfc-gh-nkrishna**
SNOW-1662275

This PR adds tests and removes `NotImplementedError` relevant to `left_index` and `right_index` parameters supplied to `pd.merge_asof`. Support was added in https://github.com/snowflakedb/snowpark-python/pull/2095

### CODE REVIEW

**Review by sfc-gh-nkumar — APPROVED** [review]

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py [hunk 1]
```diff
@@ -7381,10 +7381,10 @@ def merge_asof(
         SnowflakeQueryCompiler
         """
         # TODO: SNOW-1634547: Implement remaining parameters by leveraging `merge` implementation
-        if left_index or right_index or tolerance or suffixes != ("_x", "_y"):
+        if tolerance or suffixes != ("_x", "_y"):
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:7384]: So it was already supported? We're just removing the error messages in this PR?

Comment (sfc-gh-nkrishna) [reply src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:7384]: Yes we use a utility function as part of join, this PR just adds the tests and removes the error messages

Comment (sfc-gh-helmeleegy) [reply src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:7384]: I see. Then can you please mention that in the description and point to the original PR where the actual support was added?

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM. Thanks, Naren!

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py (hunk 1 above)
Comment (sfc-gh-nkrishna) [reply src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:7384]: Yep done

**Pull request merged**